dvc/objects/cache.py

```diff
@@ -0,0 +1,54 @@
+import pickle
+from functools import wraps
+from typing import Any
+
+import diskcache
+from diskcache import Disk as disk
+from diskcache import Index  # noqa: F401, pylint: disable=unused-import
+
+
+class DiskError(Exception):
+    def __init__(self, directory: str) -> None:
+        self.directory = directory
+        super().__init__(f"Could not open disk: {directory}")
+
+
+def translate_pickle_error(fn):
+    @wraps(fn)
+    def wrapped(self, *args, **kwargs):
+        try:
+            return fn(self, *args, **kwargs)
+        except (pickle.PickleError, ValueError) as exc:
+            if isinstance(exc, ValueError) and not str(exc).startswith(
+                "pickle protocol"
+            ):
+                raise
+            # pylint: disable=protected-access
+            raise DiskError(self._directory) from exc
+
+    return wrapped
+
+
+class Disk(disk):
+    """Reraise pickle-related errors as DiskError."""
+
+    put = translate_pickle_error(disk.put)
+    get = translate_pickle_error(disk.get)
+    store = translate_pickle_error(disk.store)
+    fetch = translate_pickle_error(disk.fetch)
+
+
+class Cache(diskcache.Cache):
+    """Extended to handle pickle errors and use a constant pickle protocol."""
+
+    def __init__(
+        self,
+        directory: str = None,
+        timeout: int = 60,
+        disk: disk = Disk,  # pylint: disable=redefined-outer-name
+        **settings: Any,
+    ) -> None:
+        settings.setdefault("disk_pickle_protocol", 4)
```

Comment:
Instead of defaulting to `pickle.HIGHEST_PROTOCOL`, this defaults to a constant value.